### PR TITLE
The Clone Wars: Attack on the Clones

### PR DESCRIPTION
**Pull request by thomastaylor312**
This PR consists of several mostly atomic commits that do some cleanup and small refactors that were much needed (it has been a while since we last cleaned up the code). Each commit explains specifically what was done and why, so I would recommend reviewing each commit separately when reviewing.

### CODE REVIEW

**Comment by thomastaylor312** [comment]
Currently doing this as a draft while I finish a few things

File: bin/server.rs [hunk 1]
```diff
@@ -153,10 +153,12 @@ async fn main() -> anyhow::Result<()> {
     // All other cases are considered errors worthy of failing.
     let keyring: KeyRing = match std::fs::metadata(&keyring_file) {
         Ok(md) if md.is_file() => load_toml(keyring_file).await?,
-        Ok(_) => Err(anyhow::anyhow!(
-            "Expected {} to be a regular file",
-            keyring_file.display()
-        ))?,
+        Ok(_) => {
+            return Err(anyhow::anyhow!(
```
Comment (technosophos) [inline bin/server.rs:157]: Can this be reduced to `anyhow::bail!`? I always forget to use `bail!()`.

File: src/invoice/mod.rs [hunk 2]
```diff
@@ -117,21 +117,21 @@ impl Invoice {
...
+        let zero_vec = Vec::with_capacity(0);
         self.parcel
-            .clone()
-            .unwrap_or_default()
+            .as_ref()
+            .unwrap_or(&zero_vec)
             .iter()
             .filter(|p| p.member_of(name))
-            .map(|p| p.clone())
+            .cloned()
```
Comment (technosophos) [inline src/invoice/mod.rs:126]: TIL....

File: src/invoice/verification.rs [hunk 3]
```diff
@@ -5,6 +5,17 @@ use tracing::{debug, info};
 
 use std::convert::TryInto;
 
+const GREEDY_VERIFICATION_ROLES: &[SignatureRole] = &[SignatureRole::Creator];
```
Comment (technosophos) [inline src/invoice/verification.rs:8]: Ah! You mentioned this! Yeah, that came out nicely.

**Review by technosophos — APPROVED** [review]

**Comment by technosophos** [comment]
Suggested: Change the PR name to "The Clone Wars: Attack on the clones"

File: bin/server.rs (hunk 1 above)
Comment (thomastaylor312) [reply bin/server.rs:157]: Oh yeah I forgot about bail. Will fix

**Pull request merged**